### PR TITLE
Feature 2 reporting system

### PR DESCRIPTION
**Pull request by nathan-1022**

### CODE REVIEW

File: src/main/java/reporting_system/BlockedUser.java [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+import data_access_storage.UserRepo;
+import data_access_storage.UserRepoManager;
+import entities.UserAccount;
+
+import javax.swing.*;
+import java.io.File;
+import java.io.IOException;
+import java.util.List;
+import java.util.Map;
+
```
Comment (cyltaracai) [inline src/main/java/reporting_system/BlockedUser.java:13]: A comment regarding what this class is about would be helpful.  I'm a bit confused about what this class does

File: src/main/java/reporting_system/BlockedUserBak.java [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+        }
+        if(flag){
+            JOptionPane.showMessageDialog(null, "block success!");
+        }else{
+            JOptionPane.showMessageDialog(null, "block fail!");
+
+        }
+        return flag;
+    }
+}
```
Comment (cyltaracai) [inline src/main/java/reporting_system/BlockedUserBak.java:53]: What's the difference between BlockedUser.java and BlockedUserBak.java?

File: src/main/java/reporting_system/BlockedUserInteractor.java [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+                String email1 = userAccount.getEmail();
+                if(email1.equals(email)){
+                    List<String> blockedAccounts = userAccount.getBlockedAccounts();
+                    blockedAccounts.add(blockedEmail);
+                    this.userRepoManager.save(requestModel2);
+                    blockedStatus = true;
+                }
+            }
+
+        }
```
Comment (cyltaracai) [inline src/main/java/reporting_system/BlockedUserInteractor.java:63]: If you have time, this part of code can be simplified by helper methods in UserAccount and UserRepo

File: src/main/java/reporting_system/ReportedUser.java [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+import data_access_storage.RequestModel;
+import data_access_storage.UserRepo;
+import data_access_storage.UserRepoManager;
+import entities.UserAccount;
+
+import javax.swing.*;
+import java.io.File;
+import java.io.IOException;
+import java.util.Map;
+
```
Comment (cyltaracai) [inline src/main/java/reporting_system/ReportedUser.java:12]: Similar to BlockedUser.java, a comment for this class would be helpful for readers to understand what it does.

File: src/main/java/reporting_system/ReportedUserInteractor.java [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+            requestModel2 = allUserAccount.get(s);
+            if(requestModel2!=null && requestModel2.getUserAccount()!=null){
+                UserAccount userAccount = requestModel2.getUserAccount();
+                String email1 = userAccount.getEmail();
+                if(email1.equals(repotedEmail)){
+                    userAccount.isReported();
+                    this.userRepoManager.save(requestModel2);
+                    reportedStatus = true;
+                }
+            }
```
Comment (cyltaracai) [inline src/main/java/reporting_system/ReportedUserInteractor.java:60]: Similar to the case of blocking users, this part of code can be simplified and comments should be added for the readers to follow your code

**Review by cyltaracai — APPROVED** [review]
1. Violations of clean architecture are not found. 
2. Code as identified in the review in BlockedUserInteractor.java and ReportedUserInteractor.java can be simplified to avoid smelly issues. 
3. Some part of code needs additional comments so that readers can follow the code. 
I'll approve this pull request so that we can start working on merging the features. But still, please address the above 3 items. Thank you

**Review by Chao-Ma-910 — DISMISSED** [review]

**Review by RitatataVan — DISMISSED** [review]

**Review by RitatataVan — COMMENTED** [review]
need to resolve conflicts

**Comment by cyltaracai** [comment]
This branch is re-created and its content is merged to the main branch via the re-created branch.

**Pull request closed without merging**